agent/tsunami/factory/preapre_tagets_tools.py

```diff
@@ -0,0 +1,96 @@
+"""Module for preparing tsunami targets."""
+import dataclasses
+import ipaddress
+import logging
+from typing import Optional, Any
+from urllib import parse
+
+from ostorlab.agent.message import message as msg
+
+
+@dataclasses.dataclass
+class Target:
+    """Data Class for tsunami target."""
+
+    address: Optional[str] = None
+    version: Optional[str] = None
+    ip_network: Optional[ipaddress.IPv4Network | ipaddress.IPv6Network] = None
+    domain: Optional[str] = None
+    url: Optional[str] = None
+
+
+def _get_schema(message: msg.Message, args: dict[str, Any]) -> str:
+    """Returns the schema to be used for the target."""
+    if message.data.get("schema") is not None:
+        return str(message.data["schema"])
+    elif message.data.get("protocol") is not None:
+        return str(message.data["protocol"])
+    elif args.get("https") is True:
+        return "https"
+    else:
+        return "http"
+
+
+def _prepare_domain_target(message: msg.Message, args: dict[str, Any]) -> list[Target]:
+    target = str(message.data["name"])
+    schema = _get_schema(message, args)
+    port = message.data.get("port")
+    if schema == "https" and port not in [443, None]:
+        url = f"https://{target}:{port}"
+    elif schema == "https":
+        url = f"https://{target}"
+    elif port == 80:
+        url = f"http://{target}"
+    elif port is None:
+        url = f"{schema}://{target}"
+    else:
+        url = f"{schema}://{target}:{port}"
+    return [Target(domain=url, url=url)]
```

Comment:
This will throw `local variable 'url' referenced before assignment` if no check hitted

Reply:
Yeah but that's impossible, the url is defined in every condition. Linters can actually detect that.